include/xtensor-io/xio_binary.hpp

```diff
@@ -22,12 +22,10 @@ namespace xt
         template <typename T>
         inline xt::svector<T> load_bin_file(std::istream& stream, bool as_big_endian)
         {
-            stream.seekg(0, stream.end);
-            auto uncompressed_size = static_cast<std::size_t>(stream.tellg());
-            stream.seekg(0, stream.beg);
-            xt::svector<T> uncompressed_buffer(uncompressed_size / sizeof(T));
-            char* buffer = reinterpret_cast<char*>(uncompressed_buffer.data());
-            stream.read(buffer, (std::streamsize)uncompressed_size);
+            std::string buffer{std::istreambuf_iterator<char>{stream}, {}};
+            std::size_t uncompressed_size = buffer.size() / sizeof(T);
+            xt::svector<T> uncompressed_buffer(uncompressed_size);
+            std::copy((const T*)(buffer.data()), (const T*)(buffer.data()) + uncompressed_size, uncompressed_buffer.begin());
```

Comment:
This creates two copies, one in `buffer` and one in `uncompressed_buffer`, but I could not make `istreambuf_iterator` work directly with `xt::svector`.
Any idea?

Reply:
Maybe something like:
```cpp
auto it = std::istreambuf_iterator<char>(stream);
std::transform(std::istreambuf_iterator<char>(stream), (std::istreambuf_iterator<char>()), std::back_inserter(uncompressed_buffer), [&it](char c) 
{
    char tmp[sizeof(T)];
    tmp[0] = c;
    std::copy_n(++it, sizeof(T)-1, tmp+1);
    T* res = reinterpret_cast<T*>(tmp);
    return res;
});
```

But I'm not really sure that this is really readable ;)

Reply:
Ok, let's keep the copy for now then.